projects/graph/src/graph.py

```diff
@@ -2,8 +2,109 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 
-
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+
+    def add_vertex(self, vertex):
+# if key doesn't exsist in the keys we create above then add it to the end of the dictionary. This should just maybe do a "'6': set()" for expample ^set command
+        if vertex.name in self.vertices:
+            raise Exception("ALREADY EXSISTS!")
+
+        self.vertices[vertex.name] = vertex
+        # pass
```

Comment:
As discussed during lecture, an alternative to this is to simply pass a string name for the vertex, and invoke the `Vertex' class' constructor function to make a Vertex object from that. It's up to you if you want to modify your code to accommodate that or keep the course we have now.